src/superphot_plus/model/data.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from typing import List
+
+import numpy as np
+from torch.utils.data import TensorDataset
+
+
+@dataclass
+class TrainData:
+    """Class that holds the data to train and validate."""
+
+    train_dataset: TensorDataset
+    val_dataset: TensorDataset
+
+    def __iter__(self):
+        return iter(
+            (
+                self.train_dataset,
```

Comment:
I do not understand how the iterator is working here. Does python iterate through the different entries independently? In parallel?

Reply:
I was using this built-in iterator to deconstruct the object as in: `train_dataset, val_dataset = train_data`, and not for iterating over the datasets. Please let me know if this looks like a bad practice and I can revert it!